examples/dreamcast/raylib/tetris/src/tetris.cpp

```diff
@@ -0,0 +1,96 @@
+#include <kos/dbgio.h>
+#include <dc/sound/stream.h>
+#include <wav/sndwav.h>
+
+double lastUpdateTime = 0;
+
+bool EventTriggered(double interval){
+    double currentTime = GetTime();
+    if(currentTime - lastUpdateTime >= interval){
+        lastUpdateTime = currentTime;
+        return true;
+    }
+    return false;
+}
+
+cd cdManager = cd();
+
+int main(){
+    const int screenWidth = 640;
+    const int screenHeight = 480;
+    wav_stream_hnd_t bgm;
+
+    /* If start, A, B, X, Y are pressed, exit the app. */
+    cont_btn_callback(0, CONT_START | CONT_A | CONT_B | CONT_X | CONT_Y,
+                      (cont_btn_callback_t)arch_exit);
+
+    InitWindow(screenWidth, screenHeight, "Tetris in KOS!");
+    SetTargetFPS(60);
+
+    snd_stream_init();
+    wav_init();
+
+    bgm = wav_create("/rd/assets/sound/bgm.adpcm", 1);
+    wav_volume(bgm, 255);
+    wav_play(bgm);
+
+    Game game = Game();
+
+    int TextUIDistance = Constants::gridWidthWithOffset + UIPadding::large;
+    int scorePaddingHeight = UIPadding::medium;
+    int scoreBoxPaddingHeight = scorePaddingHeight + 15 + UIPadding::small;
+    int nextPaddingHeight = scoreBoxPaddingHeight + 60 + UIPadding::large;
+    int nextBoxPaddingHeight = nextPaddingHeight + UIFont::medium + UIPadding::small;
+    int gameOverPaddingHeight = nextBoxPaddingHeight + 180 + UIPadding::large;
+
+    while(1){
+        game.HandleInput();
+        if(EventTriggered(0.2)){
+            game.MoveBlockDown();
+        }
+        BeginDrawing();
+        ClearBackground(darkBlue);
+        game.Draw();
+        DrawText("Hold", Constants::gridOffset - UIPadding::medium * 4, nextPaddingHeight, UIFont::medium, WHITE);
+        DrawRectangleRounded({UIPadding::medium, (float)nextBoxPaddingHeight, Constants::gridOffset - UIPadding::large, 170}, 0.3, 6, lightBlue);
+
+        DrawText("Score", TextUIDistance, scorePaddingHeight, UIFont::medium, WHITE);
+        DrawRectangleRounded({Constants::gridWidthWithOffset + UIPadding::medium, (float)scoreBoxPaddingHeight, 170, 60}, 0.3, 6, lightBlue);
+
+        char scoreText[10];
+        sprintf(scoreText, "%d", game.score);
+        Vector2 textSize = MeasureTextEx(GetFontDefault(), scoreText, UIFont::medium, 0);
+
+        DrawText(scoreText, TextUIDistance + (170 - textSize.x)/2, scoreBoxPaddingHeight + UIPadding::medium, UIFont::medium, WHITE);
+
+        DrawText("Next", TextUIDistance,  nextPaddingHeight, UIFont::medium, WHITE);
+        DrawRectangleRounded({Constants::gridWidthWithOffset + UIPadding::medium, (float)nextBoxPaddingHeight, 170, 180}, 0.3, 6, lightBlue);
+        if(game.gameOver){
+            DrawText("GAME OVER\nPress start!", TextUIDistance, gameOverPaddingHeight, UIFont::medium, WHITE);
+        }
+        game.DrawNext(TextUIDistance - 20, nextBoxPaddingHeight + UIPadding::large * 1.5);
+        
+        game.DrawHeld(-20, nextBoxPaddingHeight + UIPadding::large * 1.5);
+        EndDrawing();
+        if(!cdManager.checkStatus()){
```

Comment:
Not very good idea call it every frame.
This call locks g1_ata_mutex that in general locks filesystems too.
In this case, it interferes with sound stream.
Call it once in a second or something.